plugins/inputs/rabbitmq/rabbitmq.go

```diff
@@ -659,20 +746,51 @@ func (r *RabbitMQ) createQueueFilter() error {
 	return nil
 }
 
-func (r *RabbitMQ) shouldGatherExchange(exchange Exchange) bool {
+func (r *RabbitMQ) shouldGatherExchange(exchangeName string) bool {
 	if len(r.Exchanges) == 0 {
 		return true
 	}
 
 	for _, name := range r.Exchanges {
-		if name == exchange.Name {
+		if name == exchangeName {
 			return true
 		}
 	}
 
 	return false
 }
 
+func (r *RabbitMQ) shouldGatherFederationLink(link FederationLink) bool {
```

Comment:
I didn't include the "federation" prefix in the variable names internally because I could not think of anything that comes included in RabbitMQ that (at least at the moment) uses the term "upstream" to mean something different. I did include it in the config option names however, because it is conceivable that some new type of "upstream" is introduced in future and having the prefix makes it easier to maintain backwards compatibility in that case.